strings/base_types.h

```diff
@@ -36,22 +36,22 @@ namespace winrt::impl
         }
         else 
         {
-            abort();
+            throw std::invalid_argument("Character is not a hexadecimal digit");
```

Comment:
Can we put the character in the error message?

Reply:
The minor complication is that it could be a `char` or a `wchar_t`... likewise for the string in your other comment.

Reply:
Ah, fair enough. Basic ASCII for char and wchar_t is the same, but if random garbage ends up being passed, this might not be desirable